lib/Haskell/Law/List.agda

```diff
@@ -0,0 +1,19 @@
+module Haskell.Law.List where
+
+open import Haskell.Prim
+open import Haskell.Prim.List
+
+--------------------------------------------------
+-- ++
+
+++-[] : ∀ (xs : List a) → xs ++ [] ≡ xs
+++-[] [] = refl
+++-[] (x ∷ xs) rewrite ++-[] xs = refl
+
+[]-++ : ∀ (xs : List a) → xs ++ [] ≡ xs
+[]-++ [] = refl
+[]-++ (x ∷ xs) rewrite []-++ xs = refl
+
+++-assoc : ∀ (xs ys zs : List a) → (xs ++ ys) ++ zs ≡ xs ++ (ys ++ zs)
+++-assoc [] ys zs = refl
+++-assoc (x ∷ xs) ys zs rewrite ++-assoc xs ys zs = refl
```

Comment:
Shouldn't these be directly given as the semigroup and monoid laws in the first place?

Reply:
These are directly for the operation on lists defined in `Haskell.Prim.List`, so I think it makes sense to also have them separately and they are then used in the semigroup and monoid law proofs.